docs/tutorial.md

```diff
@@ -0,0 +1,721 @@
+
+   ![](images/view-editors.gif)
+
+4. In the storyboard, select the **Sign into Blockstack** button.
+
+5. Control-drag from the button to the code display in the editor on the right, stopping the drag at the line below controller's opening statement.
+
+   ![](images/add-action.gif)
+
+6. Repeat this process with the storyboard's purple **hello-blockstack-ios** label, and the **Sign out** button.
+
+    When you are done, your ViewController file contains the following variables:
+
+    ```swift
+    class ViewController: UIViewController {
+
+    @IBOutlet var nameLabel: UILabel!
+    @IBOutlet var signInButton: UIButton!
+    @IBOutlet var signOutButton: UIButton!
+    ```
+
+    And XCode has added two outlines to the `Main.storyboard` source.
+
+    ```xml
+    <connections>
+    <outlet property="nameLabel" destination="9eE-ZS-LU9" id="Ahv-Te-ZZo"/>
+    <outlet property="signInButton" destination="Lfp-KX-BDb" id="yef-Jj-uPt"/>
+    <outlet property="signOutButton" destination="k0H-mY-bo5" id="Y47-Za-lSc"/>
+    </connections>
+    ```
+
+    Your connectors will have their own `destination` and `id` values.
+
+
+### Edit the ViewController.swift file
+
+Now, you are ready to connect your applicaiton with your Blockstack Web
+Application. Normally, after building your Web application you would have
+registred it with Blockstack and the app would be available on the Web. This
+example skips this registroation step and uses an example application we've
+already created for you:
+
+`https://heuristic-brown-7a88f8.netlify.com/redirect.html`
+
+This web application already has a redirect in place for you. You'll reference
+this application in your mobile add for now. In XCode, do the following;
+
+
+1. Open the `ViewController.swift` file.
+2. Add an import both for `Blockstack` and for `SafariServices`.
+
+   ```
+	 import UIKit
+	 import Blockstack
+	 import SafariServices
+	 ```
+
+4. Just before the `didReceiveMemoryWarning` function a private `updateUI` function.
+
+   This function takes care of loading the user data from Blockstack.
+
+   ```
+   private func updateUI() {
+     DispatchQueue.main.async {
+    		if Blockstack.shared.isSignedIn() {
+    			// Read user profile data
+    			let retrievedUserData = Blockstack.shared.loadUserData()
+    			print(retrievedUserData?.profile?.name as Any)
+    			let name = retrievedUserData?.profile?.name ?? "Nameless Person"
+    			self.nameLabel.text = "Hello, \(name)"
+    			self.nameLabel.isHidden = false
+            self.signInButton.isHidden = true
+            self.signOutButton.isHidden = false
+    		} else {
+            self.signInButton.isHidden = false
+            self.signOutButton.isHidden = true
+            self.nameLabel.isHidden = true
+    		}
+    	}
+    }
+    ```
+
+5. Replace the content of the `viewDidLoad()` function so that it calls this private function.
+
+   ```
+	 override func viewDidLoad() {
+		self.updateUI()
+   }
```

Comment:
super.viewDidLoad() should be here before self.updateUI(). Also, the formatting is off.